src/main/java/io/projectriff/invoker/FunctionConfiguration.java

```diff
@@ -265,18 +270,51 @@ public Object create(String type) {
 					.overrideThreadContextClassLoader(functionClassLoader);
 			AutowireCapableBeanFactory factory = context.getAutowireCapableBeanFactory();
 			try {
+				Object result = null;
 				if (this.runner != null) {
-					return this.runner.getBean(type);
+					result = this.runner.getBean(type);
+				}
+				if (result == null) {
+					logger.info("No main class provided. Instantiating: " + type);
+					if (ClassUtils.isPresent(type, functionClassLoader)) {
+						result = factory.createBean(
+								ClassUtils.resolveClassName(type, functionClassLoader));
+					}
+				}
+				if (result != null) {
+					return result;
 				}
-				logger.info("No main class provided. Instantiating: " + type);
-				return factory.createBean(
-						ClassUtils.resolveClassName(type, functionClassLoader));
+				throw new IllegalStateException("Cannot create bean for: " + type);
 			}
 			finally {
 				ClassUtils.overrideThreadContextClassLoader(contextClassLoader);
 			}
 		}
 
+		/**
+		 * If a Supplier has been registered (as opposed to a Function), convert it to a
+		 * Function in such a way that it will send messages to the output stream as soon
+		 * as the rpc call starts, discarding inputs.
+		 */
+		private Object function(FunctionRegistration<Object> reg) {
+			Object result = reg.getTarget();
+			if (result instanceof Supplier) {
+				FunctionType type = reg.getType();
+				if (type == null) {
+					type = FunctionType.of(result.getClass());
+				}
+				// If we don't supply a Flux already, let's make sure we get one
+				if (!type.isWrapper()) {
+					result = new FluxSupplier<>((Supplier<?>) result);
+				}
```

Comment:
@dsyer tells me that if `isWrapper` returns `true`, then `type` must be a `Supplier<Publisher<?>>`.